corona-warn-app/cwa-app-android
# Refactor home-screen and tracing details fragment (EXPOSUREAPP-4049)

* Use `RecyclerView` for dynamic display of list items
* Remove nested/included layouts and make Android Studio layout previews functional
* Split list items into independent elements (positive/negative/pending)
* Layout cleanups & flattening
* Small padding and alignment fixes

## TODO
- [x] Fix progress behavior in tracing details view
- [x] Cleanup state providers in the home and tracing details viewmodel
- [x] Check if we can make the tracing status bar on the home screen use scroll behavior

## Testing
* Check the UI states on the home screen and tracing details screen against FIGMA mockups
* Check the conditional logic in `SubmissionStateProvider`
* Check the conditional logic in `TracingStateProvider`
